units/src/amount/mod.rs

```diff
@@ -38,6 +38,48 @@ pub use self::{
     unsigned::Amount,
 };
 
+/// Result of an operation on [`Amount`] or [`SignedAmount`].
+///
+/// The type parameter `T` should be normally `Amout` or `SignedAmount`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AmountOpResult<T> {
+    /// Result of a successful mathematical operation.
+    Valid(T),
+    /// Result of an unsuccessful mathematical operation.
+    Error(AmountOpError),
+}
+
+impl<T> AmountOpResult<T> {
+    /// Returns the contained valid amount.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the result is an `Error`.
+    pub fn unwrap_valid(self) -> T {
+        use AmountOpResult::*;
+
+        match self {
+            Valid(amount) => amount,
+            Error(_) => panic!("TODO"),
+        }
+    }
+
+    /// Converts this `AmountOpResult` to a `Result<T, AmountOpError>`.
+    pub fn valid(self) -> Result<T, AmountOpError> {
```

Comment:
I was originally thinking of naming this `into_result` which feels more readable/idiomatic to me but this one is for sure shorter. I'm not sure which is better.

Reply:
Yes, I don't think there would be a reason to do it otherwise.